grid_map_loader/src/GridMapLoader.cpp

```diff
@@ -7,21 +7,29 @@
  *
  */
 
-#include <grid_map_loader/GridMapLoader.hpp>
-#include <grid_map_msgs/GridMap.h>
+#include "grid_map_loader/GridMapLoader.hpp"
+#include <grid_map_msgs/msg/grid_map.hpp>
 
-using namespace grid_map;
+#include <chrono>
 
-namespace grid_map_loader {
+#include <string>
 
-GridMapLoader::GridMapLoader(ros::NodeHandle nodeHandle)
-    : nodeHandle_(nodeHandle)
+namespace grid_map_loader
+{
+
+GridMapLoader::GridMapLoader()
+: Node("grid_map_loader")
 {
   readParameters();
-  publisher_ = nodeHandle_.advertise<grid_map_msgs::GridMap>(publishTopic_, 1, true);
-  if (!load()) return;
+
+  rclcpp::QoS custom_qos(rclcpp::KeepLast(1));  // Buffer only last published message.
+  custom_qos.transient_local();  // Persist messages for “late-joining” subscriptions.
+  custom_qos.lifespan(duration_);  // Expiration duration of persisted messasges.
+
+  publisher_ = this->create_publisher<grid_map_msgs::msg::GridMap>(publishTopic_, custom_qos);
```

Comment:
I think the QoS should be configurable if they want one way or another. We do this in Nav2 / slam libraries too because some applications aren't compatible with this QoS

Reply:
By configurable you mean exposing them as parameters? The custom QOS implemented in nav2 that I managed to find were not configurable as far as I can tell (they were the ones in nav2_costmap_2d), can you please point me to an example?

Looking at the [QOS compatibilities tables](https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings/#qos-compatibilities) I don't see that the QOS config used here will cause any incompatibilities, as far as I can tell.

Reply:
https://github.com/ros-planning/navigation2/blob/master/nav2_costmap_2d/plugins/static_layer.cpp#L87 since this takes in the map which SLAM or localization could have different QoS settings. Its just helpful for the case you have a subscriber to these grid maps that you need to give to another process / node

Reply:
Should be configurable now. I have made the lifespan configurable only when a value bigger than 0  is passed to the `duration` param, If the user wants to keep the default QOS message lifespan, they will need to set the duration parameter to -1.